src/cldfviz/templates/text/util.md

```diff
@@ -18,8 +14,27 @@
 ){% endif %}{% endif %}
 {%- endmacro %}
 
+
 {% macro form(f, with_language=False) -%}
-{% if with_language %}{{ f.language.name }} {% endif %}_{{ f.cldf.form }}_ ‘{{ f.parameter.name if f.parameter else f.cldf.parameterReference }}’
+{% if f.cldf.form is string %}
+    {% set form_string = "_" + f.cldf.form + "_" %}
+{% else %}
+    {% set form_strings = [] %}
+    {% for form in f.cldf.form %}
+        {% set form_strings = form_strings.append("_" + form + "_") %}
+    {% endfor %}
+    {% set form_string = form_strings|join('/') %}
+{% endif %}
+{% if f.cldf.parameterReference is not string %} 
+    {% set translations = [] %}
+    {% for par in f.parameters %}
+        {% set translations = translations.append(par.name) %}
+    {% endfor %}
+    {% set trans_string = translations|join(', ') %}
+{% else %}
+    {% set trans_string = f.parameter.name %}
+{% endif %}
+{% if with_language %}{{ f.language.name }} {% endif %}{{form_string}} ‘{{ trans_string }}’
```

Comment:
@fmatter It's good to see we can make this edge case work. But I think having both - multivalued `form` and multivalued `parameterReference` - should be discouraged (and I'm a bit surprised that `"dc:extent": "singlevalued"` constraint together with a `separator` in the spec for `form` isn't flagged as invalid by `cldf validate`), because it leaves too much room for interpretation: Do all forms have all meanings - or each a subset? So in particular in cases where the CLDF is created from a separate curation format, I'd insist on duplicating things to resolve this ambiguity.

Reply:
It does raise a warning with `Wordlist`, hence my use of `Generic` :)

To me, this representation is perfectly unambiguous: there are two related forms with the same meanings, pure allomorphy and pure polysemy (although there can be edge cases).

But this is something I've been struggling with in many of my CLDF endeavours. Since I often code descriptive content, this "edge" case arises quite a lot. For example: in a comparative context, we have a 1-to-1 form-meaning pair:

| ID | Form | Parameter_ID | Language_ID | Segments
| :-------------- | :--------- | :---------- | :---------- | :----- |
| tri-bone | jetɨpə | bone | trio1238 | j e t ɨ p ə

But if we look at this word from a language-specific perspective (or a more fine-grained comparative perspective), there is both allomorphy and polysemy, i.e. a many-to-many relationship.

| ID | Form | Parameter_ID | Language_ID |
| :-------------- | :--------- | :---------- | :---------- |
| tri-bone | jetɨpə; jehpə | bone; cob | trio1238 |

I am mainly unsure about how allomorphy should be encoded. I see the 'one form, one cell' approach, but how do we related allomorphs to eachother?